### PR TITLE
Pe 6324 more robust init tests

### PR DESCRIPTION
**Pull request by ilias1111**
## Description

Adding some tests, so if the context flag is true we examine early on if we have all the required fields in the dwh for those flags

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Checklist

- [ ] 💣 Is your change a breaking change?
- [ ] 📖 I have updated the CHANGELOG.md

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📓 internal package docs (ymls, macros, readme, if applicable)
- [ ] 📕 I have raised a [Snowplow documentation](https://github.com/snowplow/documentation) PR if applicable (Link here)
- [ ] 🙅 no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

Alt Text

### CODE REVIEW

File: macros/context_existance_check.sql [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+Copyright (c) 2023-present Snowplow Analytics Ltd. All rights reserved.
+This program is licensed to you under the Snowplow Personal and Academic License Version 1.0,
+and you may not use this file except in compliance with the Snowplow Personal and Academic License Version 1.0.
+You may obtain a copy of the Snowplow Personal and Academic License Version 1.0 at https://docs.snowplow.io/personal-and-academic-license-1.0/
+#}
+
+
+{% macro context_existance_check() %}
+  {% set contexts = [
+    ['snowplow__enable_mobile_context','contexts_com_snowplowanalytics_snowplow_mobile_context_1', 'snowplow__mobile_context'],
```
Comment (agnessnowplow) [inline macros/context_existance_check.sql:11]: The issue with this currently is that we tie the redshift/postgres context tables as vars I believe specifically to be able to overwrite them for integration tests which are running against the _stg tables, we need to allow for that to work. Can you try replacing them as vars?

File: macros/context_existance_check.sql [hunk 2]
```diff
@@ -9,7 +9,7 @@ You may obtain a copy of the Snowplow Personal and Academic License Version 1.0
 {% macro context_existance_check() %}
   {% set contexts = {
       "snowplow__enable_mobile_context": [
-          'contexts_com_snowplowanalytics_snowplow_mobile_context_12' if target.type not in ['redshift'] else var('snowplow__mobile_context')
+          'contexts_com_snowplowanalytics_snowplow_mobile_context_1' if target.type not in ['redshift'] else var('snowplow__mobile_context')
```
Comment (agnessnowplow) [inline macros/context_existance_check.sql:12]: this should be redshift and postgres?

Comment (ilias1111) [reply macros/context_existance_check.sql:12]: aaa yeah!! should i try that?

File: macros/context_existance_check.sql [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+{% macro default__context_existance_check(contexts) %}
+
+  {% if execute %}
+    {%- if flags.WHICH in ('run', 'run-operation') -%}
+
+      {% set relation = adapter.get_relation(
+          database=target.database,
+          schema=var('snowplow__atomic_schema', 'atomic'),
+          identifier=var('snowplow__events_table', 'events'))
```
Comment (rlh1994) [inline macros/context_existance_check.sql:68]: We need to use the correct database which might not be the target database.

Comment (ilias1111) [reply macros/context_existance_check.sql:68]: Ok we need to it in way by explicitly stating the db, I am pretty sure we need to define the relation ship with that way

Comment (ilias1111) [reply macros/context_existance_check.sql:68]: Maybe `this.database` ?

**Comment by ilias1111** [comment]
databricks passed locally

File: integration_tests/.scripts/integration_test.sh [hunk 4]
```diff
@@ -23,38 +23,44 @@ fi
...
+  # if [ $status -ne 0 ] && [ $status -ne 1 ]; then
+  #   echo "dbt command failed for target $db with exit status $status"
+  #   exit $status
     
-  fi
+  # fi
 
   echo "Snowplow unified integration tests: Seeding data"
-  eval "dbt seed --full-refresh --target $db" || exit 1;
+  eval "dbt seed --full-refresh --target $db --no-partial-parse" || exit 1;
```
Comment (agnessnowplow) [inline integration_tests/.scripts/integration_test.sh:38]: Didn't we see that the --no-partial-parse did not work as a solution? Should we remove them not to clutter things?

File: integration_tests/.scripts/integration_test.sh [hunk 5]
```diff
@@ -23,38 +23,44 @@ fi
...
 
   echo "Snowplow unified integration tests: Try run without data"
-  eval "dbt run --full-refresh --vars '{snowplow__allow_refresh: true, snowplow__backfill_limit_days: 1, snowplow__enable_cwv: false, snowplow__start_date: 2010-01-01}' --target $db" || exit 1;
-
+  eval "dbt run --select snowplow_unified_integration_tests.source --full-refresh --vars '{snowplow__allow_refresh: true, snowplow__backfill_limit_days: 1, snowplow__enable_cwv: false, snowplow__start_date: 2010-01-01}' --target $db --no-partial-parse" || exit 1;
+  eval "dbt run --full-refresh --vars '{snowplow__allow_refresh: true, snowplow__backfill_limit_days: 1, snowplow__enable_cwv: false, snowplow__start_date: 2010-01-01}' --target $db --no-partial-parse" || exit 1;
```
Comment (agnessnowplow) [inline integration_tests/.scripts/integration_test.sh:42]: what is this extra run for?

File: integration_tests/.scripts/integration_test.sh [hunk 6]
```diff
@@ -23,38 +23,44 @@ fi
...
 
   echo "Snowplow unified integration tests: Conversions"
-  eval "dbt run --full-refresh --select +snowplow_unified_conversions snowplow_unified_integration_tests.source --vars '{snowplow__allow_refresh: true, snowplow__backfill_limit_days: 220, snowplow__enable_cwv: false, snowplow__enable_conversions: true}' --target $db" || exit 1;
+  eval "dbt run --full-refresh --select snowplow_unified_integration_tests.source --vars '{snowplow__allow_refresh: true, snowplow__backfill_limit_days: 220, snowplow__enable_cwv: false, snowplow__enable_conversions: true}' --target $db --no-partial-parse" || exit 1;
```
Comment (agnessnowplow) [inline integration_tests/.scripts/integration_test.sh:45]: I don't think we need to add vars just run the .source bits? (here and below, to reduce extra code duplication)

Comment (ilias1111) [reply integration_tests/.scripts/integration_test.sh:45]: Should we add them so the source files are created "dynamically" correctly with the right conditions?

File: integration_tests/.scripts/integration_test.sh (hunk 4 above)
Comment (ilias1111) [reply integration_tests/.scripts/integration_test.sh:38]: Will remove them!

File: integration_tests/.scripts/integration_test.sh [hunk 7]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 # Expected input:
-# -d (database) target database for dbt
+# (database) target database for dbt
```
[the remark this answers is not in the document]
Comment (ilias1111) [reply integration_tests/.scripts/integration_test.sh:4]: Done

File: integration_tests/.scripts/integration_test.sh (hunk 5 above)
Comment (agnessnowplow) [reply integration_tests/.scripts/integration_test.sh:42]: Yep, I take it back, the long list of vars tricked me for a second.

**Review by agnessnowplow — APPROVED** [review]
LGTM (guessing it would be easier at this point to just do a merge commit which we can just fix later in the release branch)

**Pull request merged**